### ISSUE TITLE
AWS Client Error - Code: InvalidArgument

### ISSUE THREAD

**Issue by alexrubin** [putyourlightson/craft-blitz-cloudfront#8]
The Blitz log is capturing the following error when I attempt to save the keys and distribution ID. 

AWS Client Error - Code: InvalidArgument - Message: Your request contains one or more invalid invalidation paths.

Let me know if there's anything more I can provide.

Blitz: v4.2.3
Blitz CloudFront: v3.0.0
Craft CMS: v4.3.0

**Comment by bencroker** [comment]
Can you add the following line after the error is logged, so we can try to identify what paths are causing issues?

```php
Blitz::$plugin->log(implode(PHP_EOL, $paths), [], Logger::LEVEL_ERROR);
```

**Comment by alexrubin** [comment]
It returns an empty line sadly.

```
[2022-11-01 17:47:13] AWS Client Error - Code: InvalidArgument - Message: Your request contains one or more invalid invalidation paths.
[2022-11-01 17:47:13] 

```

**Comment by bencroker** [comment]
Hmm, perhaps the `purgeUrisWithProgress` method needs this at the top. Care to test it out?

```php
if (empty($siteUris)) {
    return;
}
````

**Comment by alexrubin** [comment]
I was able to debug the issue. The test() function doesn't include a path and that causes the createInvalidation method on line 223 of the _sendRequest to fail because Paths > Items is required but test sends an empty array.

Update line 181 to include the global path in the parameters allows the settings to be saved.

```
/**
 * @inheritdoc
 */
public function test(): bool
{
    $response = $this->_sendRequest(['/*']);

    if (!$response) {
        return false;
    }

    return true;
}
```

**Comment by bencroker** [comment]
That makes sense. We should probably use a test path so as to not invalidate the entire cache, I'll release a fix shortly.

**Comment by bencroker** [comment]
Fixed in https://github.com/putyourlightson/craft-blitz-cloudfront/commit/cbc707cbabf7eb4fa5a756dbbf5f75c6842e7e02 and released in 3.0.2.
